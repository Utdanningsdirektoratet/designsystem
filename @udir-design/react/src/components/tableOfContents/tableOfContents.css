.uds-table-of-contents {
  width: 100%;

  & .ds-link {
    --dsc-link-color--visited: var(--dsc-link-color);
    text-decoration: none;
  }

  & ol {
    list-style-type: none;
    margin: 0;
    padding: 0;

    .ds-link {
      display: flex;
      width: fit-content;

      & svg {
        flex-shrink: 0;
        width: var(--ds-size-6);
        height: var(--ds-size-6);
        align-self: flex-start;
      }
    }

    .ds-link:hover {
      text-decoration: underline;
    }

    & li:not(:last-child) {
      margin-bottom: var(--ds-size-3);
    }
  }

  & li.level-3,
  li.level-4 {
    margin-left: var(--ds-size-3);
  }
}
