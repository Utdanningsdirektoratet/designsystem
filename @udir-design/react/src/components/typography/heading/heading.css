.ds-heading:focus {
  outline: none;
}
