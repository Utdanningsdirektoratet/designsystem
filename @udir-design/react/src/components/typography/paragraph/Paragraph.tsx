import { Paragraph } from '@digdir/designsystemet-react';

export { Paragraph };
export type { ParagraphProps } from '@digdir/designsystemet-react';
