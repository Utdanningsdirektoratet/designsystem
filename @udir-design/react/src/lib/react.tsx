import styles from './react.module.scss';

export const React: React.FC = () => (
  <div className={styles['container']}>
    <h1>Welcome to React!</h1>
  </div>
);

export default React;
