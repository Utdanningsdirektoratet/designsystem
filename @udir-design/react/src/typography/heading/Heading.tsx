export { Heading, type HeadingProps } from '@digdir/designsystemet-react';
