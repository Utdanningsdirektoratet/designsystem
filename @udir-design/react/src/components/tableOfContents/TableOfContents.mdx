import {
  Meta,
  Primary,
  Controls,
  Canvas,
  ArgTypes,
} from '@storybook/addon-docs/blocks';
import * as TableOfContentStories from './TableOfContents.stories';

<Meta of={TableOfContentStories} />

# Table Of Contents

`TableOfContents` gir brukeren bedre oversikt over innhold i lengre artikler. Komponenten inneholder lenker som scroller ned til den valgte overskriften.

**Passer til å**

- gi et overblikk på sider med mye innhold
- la brukeren navigere til overskrifter på en side

**Passer ikke til å**

- lenke til innhold som ikke er en [`Heading`](/docs/components-typography--docs)

## Slik bruker du Table Of Contents

<Primary />
<Controls />

```jsx
import { TableOfContents, useTableOfContents } from '@udir-design/react';

// med useTableOfContents()
<TableOfContents {...useTableOfContents()} />;

// med hardkodede headings
const h = [
  {
    name: 'Første overskrift',
    level: 2,
    id: 'forste-overskrift'
  },
  {
    name: 'Andre overskrift',
    level: 2,
    id: 'andre-overskrift'
  }
]
<TableOfContents headings={h} />;
```

## Eksempler

### Generere overskrifter automatisk

Hooken [`useTableOfContents`](/docs/utilities-usetableofcontents--docs) kan brukes for å automatisk fylle `TableOfContents` med alle overskrifter som har en id, i dokumentet. Du kan velge om den skal ta med kun `h2` eller både `h2` og `h3`.

Dersom det er enkelte overskrifter du ikke vil inkludere i innholdsfortegnelsen kan du bruke `data-toc-ignore` på `Heading`.

<Canvas
  of={TableOfContentStories.Automatic}
  story={{ inline: false, height: '450px' }}
/>

### Legge til overskrifter manuelt

Dersom du har behov for full kontroll over innholdet i `TableOfContents` kan du lage en liste av objekter som inneholder `name`, `level` og `id`. Id brukes til å lenke til riktig overskrift, så det er viktig at denne stemmer overens med overskriftens id i dokumentet.

<Canvas
  of={TableOfContentStories.Manual}
  story={{ inline: false, height: '450px' }}
/>

## Retningslinjer

`h1` skal ikke inngå i en `TableOfContents`.
