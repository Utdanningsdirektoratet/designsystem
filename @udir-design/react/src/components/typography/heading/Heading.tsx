import {
  Heading as DigdirHeading,
  HeadingProps,
} from '@digdir/designsystemet-react';
import { forwardRef } from 'react';
import './heading.css';

export const Heading = forwardRef<HTMLHeadingElement, HeadingProps>(
  function Heading({ children, id, ...rest }, ref) {
    return (
      <DigdirHeading
        {...rest}
        id={id}
        ref={ref}
        {...(id ? { tabIndex: -1 } : {})}
      >
        {children}
      </DigdirHeading>
    );
  },
);

export type { HeadingProps };
