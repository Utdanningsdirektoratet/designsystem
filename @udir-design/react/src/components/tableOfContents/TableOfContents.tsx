import cl from 'clsx/lite';
import { forwardRef } from 'react';
import { ArrowDownRightIcon } from '@udir-design/icons';
import { CardProps as DigdirCardProps } from '../card/Card';
import { Card } from '../card/Card';
import { Details } from '../details/Details';
import { Link } from '../link/Link';
import './tableOfContents.css';

export type TocHeading = {
  level: 2 | 3 | 4;
  name: string;
  id: string;
};

export type TableOfContentsProps = Omit<
  DigdirCardProps,
  'data-color' | 'asChild' | 'children'
> & {
  /**
   * Headings to be displayed
   * in the tableOfContents
   */
  headings: TocHeading[];
  'data-color'?: 'neutral' | 'accent' | 'support1';
};

export const TableOfContents = forwardRef<HTMLDivElement, TableOfContentsProps>(
  function TableOfContents({ headings, variant, className, ...rest }, ref) {
    return (
      <Card
        variant={variant}
        className={cl(`uds-table-of-contents`, className)}
        ref={ref}
        {...rest}
      >
        <Details defaultOpen>
          <Details.Summary>Innhold p√• denne siden</Details.Summary>
          <Details.Content>
            <ol>
              {headings.map((header) => (
                <li key={header.id} className={`level-${header.level}`}>
                  <Link href={`#${header.id}`}>
                    <ArrowDownRightIcon aria-hidden />
                    <span>{header.name}</span>
                  </Link>
                </li>
              ))}
            </ol>
          </Details.Content>
        </Details>
      </Card>
    );
  },
);
