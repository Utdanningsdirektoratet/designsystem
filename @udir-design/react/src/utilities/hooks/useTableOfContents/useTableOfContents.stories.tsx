import type { Meta, StoryObj } from '@storybook/react-vite';
import { useRef } from 'react';
import { expect, within } from 'storybook/test';
import { withScrollHashBehavior } from '.storybook/decorators/withScrollHashBehavior';
import { TableOfContents } from 'src/components/tableOfContents/TableOfContents';
import { Heading } from 'src/components/typography/heading/Heading';
import { Paragraph } from 'src/components/typography/paragraph/Paragraph';
import styles from './docs/UseTableOfContents.module.css';
import {
  useTableOfContents,
  type useTableOfContentsProps,
} from './useTableOfContents';

export default {
  title: 'Utilities/useTableOfContents',
  parameters: {
    chromatic: { disableSnapshot: true },
    tags: ['alpha', 'udir'],
    componentOrigin: {
      originator: 'self',
    },
  },
  decorators: [withScrollHashBehavior],
  argTypes: {
    containerRef: {
      table: {
        type: { summary: 'object' },
        defaultValue: { summary: 'Document' },
      },
      description: 'The RefObject of the element to extract headings from',
    },
    headingSelector: {
      description: 'Which headings to include in the TableOfContents',
      table: {
        defaultValue: { summary: 'h2,h3' },
        type: { summary: 'h2|h2,h3' },
      },
    },
  },
} as Meta;

type Story = StoryObj<useTableOfContentsProps>;

export const Default: Story = {
  render() {
    return (
      <div className={styles.prose}>
        <TableOfContents {...useTableOfContents()} data-testid="toc" />
        <Heading
          level={2}
          id="first-heading"
          data-size="md"
          data-testid="first-heading"
        >
          Første overskrift (h2)
        </Heading>
        <Heading level={3} id="first-subheading" data-size="sm">
          Første underoverskrift (h3)
        </Heading>
        <Heading level={2} id="second-heading" data-size="md">
          Andre overskrift (h2)
        </Heading>
      </div>
    );
  },
  play: async ({ canvasElement, step }) => {
    const canvas = within(canvasElement);
    const toc = canvas.getByTestId('toc');
    const links = within(toc).getAllByRole('link');

    const linkData = links.map((link) => ({
      href: link.getAttribute('href'),
      text: link.textContent?.trim(),
    }));

    await step('All three headings should be part of toc', async () => {
      expect(linkData).toEqual([
        { href: '#first-heading', text: 'Første overskrift (h2)' },
        { href: '#first-subheading', text: 'Første underoverskrift (h3)' },
        { href: '#second-heading', text: 'Andre overskrift (h2)' },
      ]);
    });
  },
};

export const Container: Story = {
  render() {
    const containerRef = useRef<HTMLDivElement>(null);
    return (
      <div ref={containerRef} className={styles.prose}>
        <Heading level={1} id="sidetittel" data-size="lg">
          Sidetittel (h1)
        </Heading>
        <Paragraph data-size="lg">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec dui
          nunc, maximus vitae finibus ut, consequat sit amet metus. Suspendisse
          nisl massa, egestas eu justo eu, luctus volutpat ante. Aliquam congue
          consectetur arcu, at pharetra nunc ultricies in. Sed aliquam mi a odio
          varius, in eleifend orci elementum.
        </Paragraph>
        <TableOfContents {...useTableOfContents({ containerRef })} />
        <Heading level={2} id="forste-overskrift" data-size="md">
          Første overskrift (h2)
        </Heading>
        <Paragraph>
          Vestibulum ut mauris quis elit hendrerit luctus. Curabitur sit amet
          mauris id nisi vehicula accumsan. Curabitur lobortis dolor sed nunc
          consectetur, vel lobortis felis rhoncus. Etiam elit dolor, convallis
          id felis in, vulputate condimentum turpis.
        </Paragraph>
        <Paragraph>
          Vestibulum volutpat augue sed dui tristique, eget accumsan massa
          laoreet. Sed consequat, ex ullamcorper tempus faucibus, nibh quam
          dignissim augue, iaculis vestibulum arcu magna id mauris. Lorem ipsum
          dolor sit amet, consectetur adipiscing elit. Duis sollicitudin massa
          id leo auctor viverra vel vitae felis. Mauris vehicula porttitor
          congue.
        </Paragraph>
        <Heading level={3} id="forste-underoverskrift" data-size="sm">
          Første underoverskrift (h3)
        </Heading>
        <Paragraph>
          Suspendisse in vulputate dui, sed rutrum arcu. Nunc vitae fermentum
          nunc, ut dignissim quam. Fusce at dolor venenatis nulla gravida
          ornare. Proin pulvinar tincidunt nisl, eu venenatis elit consequat et.
          Suspendisse blandit felis quis ante hendrerit, vitae aliquet odio
          lobortis. Vivamus at ex id magna cursus consectetur id luctus elit.
          Vivamus vitae ex vel sapien convallis commodo in quis eros. Nunc a
          volutpat urna.
        </Paragraph>
        <Heading level={4} id="forste-underunderoverskrift" data-size="xs">
          Første underunderoverskrift (h4)
        </Heading>
        <Paragraph>
          Donec at pellentesque lacus. Integer pretium, nulla vitae tincidunt
          tincidunt, mauris augue facilisis sapien, eget pulvinar odio ex
          accumsan nulla. Proin venenatis rhoncus suscipit. Proin ac ex viverra,
          laoreet enim at, tempor elit. Nunc diam eros, congue pharetra ligula
          in, luctus molestie nibh. Donec placerat ac lacus a feugiat. Sed justo
          elit, vestibulum id iaculis fermentum, ornare a nisi.
        </Paragraph>
        <Heading level={2} id="andre-overskrift" data-size="md">
          Andre overskrift (h2)
        </Heading>
        <Paragraph>
          Quisque lacinia sodales purus non lobortis. Quisque bibendum, nulla ut
          viverra tempor, nisl nibh dictum tellus, ac dapibus quam est non dui.
          Integer fringilla commodo lectus ut suscipit. Nulla at porttitor eros.
          Cras consequat ex nec pellentesque pharetra. Vivamus facilisis
          consequat velit sit amet pretium.
        </Paragraph>
      </div>
    );
  },
};

export const TocIgnore: Story = {
  render() {
    return (
      <div className={styles.prose}>
        <TableOfContents {...useTableOfContents()} />
        <Heading level={2} id="inkluder-overskrift-1" data-size="md">
          Inkluder første overskrift (h2)
        </Heading>
        <Heading
          level={2}
          id="utelat-overskrift-2"
          data-toc-ignore
          data-size="md"
        >
          Utelat andre overskrift (h2)
        </Heading>
        <Heading level={2} id="inkluder-overskrift-3" data-size="md">
          Inkluder tredje overskrift (h2)
        </Heading>
      </div>
    );
  },
};
