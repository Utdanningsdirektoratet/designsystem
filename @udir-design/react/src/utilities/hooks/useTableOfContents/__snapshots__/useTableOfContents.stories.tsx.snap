// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container 1`] = `
<div>
  <h1
    data-size="lg"
    id="sidetittel"
    tabindex="-1"
  >
    Sidetittel (h1)
  </h1>
  <p
    data-variant="default"
    data-size="lg"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec dui nunc, maximus vitae finibus ut, consequat sit amet metus. Suspendisse nisl massa, egestas eu justo eu, luctus volutpat ante. Aliquam congue consectetur arcu, at pharetra nunc ultricies in. Sed aliquam mi a odio varius, in eleifend orci elementum.
  </p>
  <div data-variant="default">
    <u-details
      open
      data-variant="default"
      role="group"
    >
      <u-summary
        aria-expanded="true"
        role="button"
        slot="summary"
        tabindex="0"
      >
        Innhold på denne siden
      </u-summary>
      <div>
        <ol>
          <li>
            <a href="#forste-overskrift">
              <svg
                xmlns="http://www.w3.org/2000/svg"
                width="1em"
                height="1em"
                fill="none"
                viewbox="0 0 24 24"
                focusable="false"
                role="img"
                aria-hidden="true"
              >
                <path
                  fill="currentColor"
                  d="M4.5 3.25a.75.75 0 0 1 .75.75v6c0 .69.56 1.25 1.25 1.25h10.69l-3.22-3.22a.75.75 0 1 1 1.06-1.06l4.5 4.5a.75.75 0 0 1 0 1.06l-4.5 4.5a.75.75 0 1 1-1.06-1.06l3.22-3.22H6.5A2.75 2.75 0 0 1 3.75 10V4a.75.75 0 0 1 .75-.75"
                >
                </path>
              </svg>
              <span>
                Første overskrift (h2)
              </span>
            </a>
          </li>
          <li>
            <a href="#forste-underoverskrift">
              <svg
                xmlns="http://www.w3.org/2000/svg"
                width="1em"
                height="1em"
                fill="none"
                viewbox="0 0 24 24"
                focusable="false"
                role="img"
                aria-hidden="true"
              >
                <path
                  fill="currentColor"
                  d="M4.5 3.25a.75.75 0 0 1 .75.75v6c0 .69.56 1.25 1.25 1.25h10.69l-3.22-3.22a.75.75 0 1 1 1.06-1.06l4.5 4.5a.75.75 0 0 1 0 1.06l-4.5 4.5a.75.75 0 1 1-1.06-1.06l3.22-3.22H6.5A2.75 2.75 0 0 1 3.75 10V4a.75.75 0 0 1 .75-.75"
                >
                </path>
              </svg>
              <span>
                Første underoverskrift (h3)
              </span>
            </a>
          </li>
          <li>
            <a href="#andre-overskrift">
              <svg
                xmlns="http://www.w3.org/2000/svg"
                width="1em"
                height="1em"
                fill="none"
                viewbox="0 0 24 24"
                focusable="false"
                role="img"
                aria-hidden="true"
              >
                <path
                  fill="currentColor"
                  d="M4.5 3.25a.75.75 0 0 1 .75.75v6c0 .69.56 1.25 1.25 1.25h10.69l-3.22-3.22a.75.75 0 1 1 1.06-1.06l4.5 4.5a.75.75 0 0 1 0 1.06l-4.5 4.5a.75.75 0 1 1-1.06-1.06l3.22-3.22H6.5A2.75 2.75 0 0 1 3.75 10V4a.75.75 0 0 1 .75-.75"
                >
                </path>
              </svg>
              <span>
                Andre overskrift (h2)
              </span>
            </a>
          </li>
        </ol>
      </div>
    </u-details>
  </div>
  <h2
    data-size="md"
    id="forste-overskrift"
    tabindex="-1"
  >
    Første overskrift (h2)
  </h2>
  <p data-variant="default">
    Vestibulum ut mauris quis elit hendrerit luctus. Curabitur sit amet mauris id nisi vehicula accumsan. Curabitur lobortis dolor sed nunc consectetur, vel lobortis felis rhoncus. Etiam elit dolor, convallis id felis in, vulputate condimentum turpis.
  </p>
  <p data-variant="default">
    Vestibulum volutpat augue sed dui tristique, eget accumsan massa laoreet. Sed consequat, ex ullamcorper tempus faucibus, nibh quam dignissim augue, iaculis vestibulum arcu magna id mauris. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis sollicitudin massa id leo auctor viverra vel vitae felis. Mauris vehicula porttitor congue.
  </p>
  <h3
    data-size="sm"
    id="forste-underoverskrift"
    tabindex="-1"
  >
    Første underoverskrift (h3)
  </h3>
  <p data-variant="default">
    Suspendisse in vulputate dui, sed rutrum arcu. Nunc vitae fermentum nunc, ut dignissim quam. Fusce at dolor venenatis nulla gravida ornare. Proin pulvinar tincidunt nisl, eu venenatis elit consequat et. Suspendisse blandit felis quis ante hendrerit, vitae aliquet odio lobortis. Vivamus at ex id magna cursus consectetur id luctus elit. Vivamus vitae ex vel sapien convallis commodo in quis eros. Nunc a volutpat urna.
  </p>
  <h4
    data-size="xs"
    id="forste-underunderoverskrift"
    tabindex="-1"
  >
    Første underunderoverskrift (h4)
  </h4>
  <p data-variant="default">
    Donec at pellentesque lacus. Integer pretium, nulla vitae tincidunt tincidunt, mauris augue facilisis sapien, eget pulvinar odio ex accumsan nulla. Proin venenatis rhoncus suscipit. Proin ac ex viverra, laoreet enim at, tempor elit. Nunc diam eros, congue pharetra ligula in, luctus molestie nibh. Donec placerat ac lacus a feugiat. Sed justo elit, vestibulum id iaculis fermentum, ornare a nisi.
  </p>
  <h2
    data-size="md"
    id="andre-overskrift"
    tabindex="-1"
  >
    Andre overskrift (h2)
  </h2>
  <p data-variant="default">
    Quisque lacinia sodales purus non lobortis. Quisque bibendum, nulla ut viverra tempor, nisl nibh dictum tellus, ac dapibus quam est non dui. Integer fringilla commodo lectus ut suscipit. Nulla at porttitor eros. Cras consequat ex nec pellentesque pharetra. Vivamus facilisis consequat velit sit amet pretium.
  </p>
</div>
`;

exports[`Default 1`] = `
<div>
  <div data-variant="default">
    <u-details
      open
      data-variant="default"
      role="group"
    >
      <u-summary
        aria-expanded="true"
        role="button"
        slot="summary"
        tabindex="0"
      >
        Innhold på denne siden
      </u-summary>
      <div>
        <ol>
          <li>
            <a href="#first-heading">
              <svg
                xmlns="http://www.w3.org/2000/svg"
                width="1em"
                height="1em"
                fill="none"
                viewbox="0 0 24 24"
                focusable="false"
                role="img"
                aria-hidden="true"
              >
                <path
                  fill="currentColor"
                  d="M4.5 3.25a.75.75 0 0 1 .75.75v6c0 .69.56 1.25 1.25 1.25h10.69l-3.22-3.22a.75.75 0 1 1 1.06-1.06l4.5 4.5a.75.75 0 0 1 0 1.06l-4.5 4.5a.75.75 0 1 1-1.06-1.06l3.22-3.22H6.5A2.75 2.75 0 0 1 3.75 10V4a.75.75 0 0 1 .75-.75"
                >
                </path>
              </svg>
              <span>
                Første overskrift (h2)
              </span>
            </a>
          </li>
          <li>
            <a href="#first-subheading">
              <svg
                xmlns="http://www.w3.org/2000/svg"
                width="1em"
                height="1em"
                fill="none"
                viewbox="0 0 24 24"
                focusable="false"
                role="img"
                aria-hidden="true"
              >
                <path
                  fill="currentColor"
                  d="M4.5 3.25a.75.75 0 0 1 .75.75v6c0 .69.56 1.25 1.25 1.25h10.69l-3.22-3.22a.75.75 0 1 1 1.06-1.06l4.5 4.5a.75.75 0 0 1 0 1.06l-4.5 4.5a.75.75 0 1 1-1.06-1.06l3.22-3.22H6.5A2.75 2.75 0 0 1 3.75 10V4a.75.75 0 0 1 .75-.75"
                >
                </path>
              </svg>
              <span>
                Første underoverskrift (h3)
              </span>
            </a>
          </li>
          <li>
            <a href="#second-heading">
              <svg
                xmlns="http://www.w3.org/2000/svg"
                width="1em"
                height="1em"
                fill="none"
                viewbox="0 0 24 24"
                focusable="false"
                role="img"
                aria-hidden="true"
              >
                <path
                  fill="currentColor"
                  d="M4.5 3.25a.75.75 0 0 1 .75.75v6c0 .69.56 1.25 1.25 1.25h10.69l-3.22-3.22a.75.75 0 1 1 1.06-1.06l4.5 4.5a.75.75 0 0 1 0 1.06l-4.5 4.5a.75.75 0 1 1-1.06-1.06l3.22-3.22H6.5A2.75 2.75 0 0 1 3.75 10V4a.75.75 0 0 1 .75-.75"
                >
                </path>
              </svg>
              <span>
                Andre overskrift (h2)
              </span>
            </a>
          </li>
        </ol>
      </div>
    </u-details>
  </div>
  <h2
    data-size="md"
    id="first-heading"
    tabindex="-1"
  >
    Første overskrift (h2)
  </h2>
  <h3
    data-size="sm"
    id="first-subheading"
    tabindex="-1"
  >
    Første underoverskrift (h3)
  </h3>
  <h2
    data-size="md"
    id="second-heading"
    tabindex="-1"
  >
    Andre overskrift (h2)
  </h2>
</div>
`;

exports[`Toc Ignore 1`] = `
<div>
  <div data-variant="default">
    <u-details
      open
      data-variant="default"
      role="group"
    >
      <u-summary
        aria-expanded="true"
        role="button"
        slot="summary"
        tabindex="0"
      >
        Innhold på denne siden
      </u-summary>
      <div>
        <ol>
          <li>
            <a href="#inkluder-overskrift-1">
              <svg
                xmlns="http://www.w3.org/2000/svg"
                width="1em"
                height="1em"
                fill="none"
                viewbox="0 0 24 24"
                focusable="false"
                role="img"
                aria-hidden="true"
              >
                <path
                  fill="currentColor"
                  d="M4.5 3.25a.75.75 0 0 1 .75.75v6c0 .69.56 1.25 1.25 1.25h10.69l-3.22-3.22a.75.75 0 1 1 1.06-1.06l4.5 4.5a.75.75 0 0 1 0 1.06l-4.5 4.5a.75.75 0 1 1-1.06-1.06l3.22-3.22H6.5A2.75 2.75 0 0 1 3.75 10V4a.75.75 0 0 1 .75-.75"
                >
                </path>
              </svg>
              <span>
                Inkluder første overskrift (h2)
              </span>
            </a>
          </li>
          <li>
            <a href="#inkluder-overskrift-3">
              <svg
                xmlns="http://www.w3.org/2000/svg"
                width="1em"
                height="1em"
                fill="none"
                viewbox="0 0 24 24"
                focusable="false"
                role="img"
                aria-hidden="true"
              >
                <path
                  fill="currentColor"
                  d="M4.5 3.25a.75.75 0 0 1 .75.75v6c0 .69.56 1.25 1.25 1.25h10.69l-3.22-3.22a.75.75 0 1 1 1.06-1.06l4.5 4.5a.75.75 0 0 1 0 1.06l-4.5 4.5a.75.75 0 1 1-1.06-1.06l3.22-3.22H6.5A2.75 2.75 0 0 1 3.75 10V4a.75.75 0 0 1 .75-.75"
                >
                </path>
              </svg>
              <span>
                Inkluder tredje overskrift (h2)
              </span>
            </a>
          </li>
        </ol>
      </div>
    </u-details>
  </div>
  <h2
    data-size="md"
    id="inkluder-overskrift-1"
    tabindex="-1"
  >
    Inkluder første overskrift (h2)
  </h2>
  <h2
    data-toc-ignore="true"
    data-size="md"
    id="utelat-overskrift-2"
    tabindex="-1"
  >
    Utelat andre overskrift (h2)
  </h2>
  <h2
    data-size="md"
    id="inkluder-overskrift-3"
    tabindex="-1"
  >
    Inkluder tredje overskrift (h2)
  </h2>
</div>
`;
